src/SchemaGenerator.ts

```diff
@@ -142,6 +142,13 @@ export class SchemaGenerator {
     }
     protected inspectNode(node: ts.Node, typeChecker: ts.TypeChecker, allTypes: Map<string, ts.Node>): void {
         switch (node.kind) {
+            case ts.SyntaxKind.VariableDeclaration: {
+                const variableDeclarationNode = node as ts.VariableDeclaration;
+                if (variableDeclarationNode.initializer?.kind === ts.SyntaxKind.ArrowFunction) {
```

Comment:
What about functions that are not arrow functions?

Reply:
I'll add a test case and see if it works.
Something like this?
```js
const myFunction = function() {};
```

Reply:
Yep

Reply:
Yep, I confirmed that function expressions do not work.
I added `ts.SyntaxKind.FunctionExpression` to the if statement, but I ran into the following error:

```

  ● valid-data-other › function-function-syntax

    Unknown node " function () {}

      45 |         }
      46 |
    > 47 |         throw new UnknownNodeError(node, context.getReference());
         |               ^
      48 |     }
      49 | }
      50 |

      at ChainNodeParser.getNodeParser (src/ChainNodeParser.ts:47:15)
      at ChainNodeParser.createType (src/ChainNodeParser.ts:32:25)
      at TopRefNodeParser.createType (src/TopRefNodeParser.ts:14:47)
      at map (src/SchemaGenerator.ts:33:40)
          at Array.map (<anonymous>)
      at SchemaGenerator.createSchemaFromNodes (src/SchemaGenerator.ts:32:14)
      at SchemaGenerator.createSchema (src/SchemaGenerator.ts:27:21)
      at Object.<anonymous> (test/utils.ts:44:34)
```

It looks like the schema generator can handle `SyntaxKind.FunctionDeclaration`, but not `SyntaxKind.FunctionExpression` which is an unsupported node type. Here's a [AST viewer](https://ts-ast-viewer.com/#code/GYVwdgxgLglg9mABKSBGAFASkQbwL4BQBECAzlMuBAEyIC8lksCiWueQA) for the difference.


While I don't think it will be difficult to support `FunctionExpression`, I feel like that's outside the scope of this PR. What do you thing?

Reply:
I'm not sure exactly what the problem is in the first place to be honest. I use functions and arrow functions in Vega-Lite and the parser doesn't seem to have any issues with it. I think we can throw our hands up for functions but we recently added support for adding function parameters to the generated schema so I'm not sure what would happen if we just ignored functions entirely. Please take a closer look at related issues and prs and see what a good behavior is.

Reply:
It seems like there are two ways to declare function types.
```
// This way is tested and works:
function myFunc1() {}

// This way is not tested and doesn't work:
const myFunc2 = function() {}
```

Since the second way of declaring functions is currently not supported, my thought is to leave functions out of this `if` statement for this pr and log an issue to add support for the second way to declare functions.

Reply:
The two ways should be equivalent so I would say either support both or none.

Reply:
I agree that both should be supported, but should adding support for the second form be in this PR or in another PR?
If it were as easy as adding a condition to the if statement, I would, but adding support for the second form means creating a new node parser.

I could add the new node parser in this PR, but the work would split really nicely into 2 PRs.

Regardless of which route we take, normal functions still work correctly since this test still passes: `test/valid-data/function-parameters-declaration/main.ts`.

What do you think?

Reply:
I see. So are you saying that named functions don't cause the parser to break? If so, then could we make the function parameter parser work with lambda functions as well instead of ignoring them here?

Reply:
Looking at how functions are parsed, it's actually pretty straightforward to add support for function expressions!
I pushed a commit to implement this.